### ISSUE TITLE
Error plaidml - Python Keras - macoS Mojave - AMD Radeon RX Vega 56

### ISSUE THREAD

**Issue by CvRNL** [plaidml/plaidml#506]
Hello all,

I experienced a problem using plaidml to train my Keras model (mostly conv2d layer and dropout layers, for image segmentation), on my MacBook Pro, macOS, Mojave, using my Blackmagic eGPU pro (AMD RAdeon RX Vega 56).

I have tried to use the OpenCl en Metal option, using the setup of plaidml through the terminal, but both give the same problem. When the first epoch should start, it gives first 

> ERROR:plaidml:Caused GPU Timeout Error (IOAF code 2)

, then a whole lot of 

> ERROR:plaidml:Internal Error (IOAF code -536870211)

Between those error lines, the epoch does seem to do something, because it also printed   

> 32/9468 [..............................] - ETA: 23:05:54 - loss: 0.6931 - acc: 0.2017

After many, many lines, it changed to error 

> ERROR:plaidml:Ignored (for causing prior/excessive GPU errors) (IOAF code 4)

Again, with a single epoch progres line in between.

It ended with 

> ERROR:plaidml:Kernel execution failure: Ignored (for causing prior/excessive GPU errors) (IOAF code 4)
> ERROR:plaidml:Ignored (for causing prior/excessive GPU errors) (IOAF code 4)

.

When looking at the GPU acitvity monitor, there were two activity lines on the external eGPU, so at some point it did communicate for a short while.

Last week, I used the eGPU for the first time, with plaidml. Then I worked on another project that just used categorical classification, and this worked fine, without errors.

Anybody any ideas on how to fix this? Thank you!

**Comment by kenthinson** [comment]
I also have a Vega 56 if you need another tester let me know.

**Comment by manchhui** [comment]
I have an AMD Radeon Pro 5300M GPU and running into similar issues when using Conv1D layer (CNN model). I will soon have a Vega 56 and will report back on if I see similar crashes.

**Comment by manchhui** [comment]
Can confirm this same issue occurs with my Vega 56... The model code is as follows:

 model = Sequential()
 model.add(InputLayer(input_shape=input_shape))
 model.add(Bidirectional(GRU(160, return_sequences=True)))
 model.add(Dropout(0.25))
 model.add(Bidirectional(LSTM(224, return_sequences=True)))
 model.add(SpatialDropout1D(0.3))
 model.add(Bidirectional(LSTM(64, return_sequences=True)))
 model.add(Dropout(0.1))
 model.add(Dense(units=1, activation = 'relu'))
 optimizer = keras.optimizers.Adam(lr=learning_rate)
 model.compile(loss=MCRMSE, metrics=['accuracy'], optimizer=optimizer)
